### ISSUE TITLE
Draw anti-aliased / smooth lines

### ISSUE THREAD

**Issue by thomasreiser** [GreycLab/CImg#373]
Is there a way to draw anti-aliased / smooth lines?

**Comment by dtschump** [comment]
No, but the [Xiaolin Wu's anti-aliased line algorithm](https://en.wikipedia.org/wiki/Xiaolin_Wu%27s_line_algorithm) is trivial to implement with CImg, and it is giving quite good results.
I've not done it in CImg, but for my other project G'MIC, and it looks OK.

![gmic_line_aa](https://user-images.githubusercontent.com/10114565/212874540-b6f09f36-0157-4e51-9857-945b5bb13bd3.png)
